test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -0,0 +1,70 @@
+// Flags: --expose-internals
+import '../common/index.mjs';
+import mod from 'internal/modules/esm/loader';
```

Comment:
I don't find `mod` very descriptive (if it's a well known acronym that I don't know, please ignore)
```suggestion
import loader from 'internal/modules/esm/loader';
```

Reply:
It's not super common. I usually use `module`, but some linters freak out when they find it in an ESM file. I considered `loader`, but it's not the loader, so I opted for `mod`

Reply:
I went with `esmLoaderModule` (not married to it).

Reply:
I dislike default exports in general. As far as I’m concerned, `esm/loader` doesn’t need a default export and you should be able to just import it via `import { loader } from 'internal/modules/esm/loader'`.

Reply:
Yeah, that's what I wanted to do. It doesn't work: https://github.com/nodejs/node/pull/41975#discussion_r807158370

> This was kind of weird btw: `esm/loader.js` exports `ESMLoader` as a prop on `exports`
>
> https://github.com/nodejs/node/blob/bbd7cf72574805281ae28486c26e7811c798016a/lib/internal/modules/esm/loader.js#L589
>
> So I expect it to be exposed as a named export (thus precluding this altogether).

`esmLoaderModule` is an object that contains a property named `ESMLoader`; there are no named exports available :(